docs/api/components.rst

```diff
@@ -0,0 +1,165 @@
+.. SPDX-License-Identifier: MIT
+
+.. currentmodule:: disnake
+
+.. _disnake_components:
+
+Message Components
+==================
+
+This section documents everything related to message components - Discord feature which allows bot developers
+to create their own component-based UIs right inside Discord.
```

Comment:
Note that these classes are not meant to be created by the user, they should use the bot ui kit components for that then link to those.

Reply:
This is already done by `|discord_model|` in the classes' docstrings, but agree, adding another warning would be better

Reply:
Done in [`d15ddd7`](https://github.com/DisnakeDev/disnake/pull/722/commits/d15ddd70e20c611070a3445379763a80b07a8c13)